File: src/main/java/rars/venus/settings/editor/controllers/PresetsController.java
package rars.venus.settings.editor.controllers;

import org.jetbrains.annotations.NotNull;
import rars.settings.SettingsThemePresets;
import rars.venus.editors.TextEditingArea;
import rars.venus.settings.editor.views.PresetsView;

public final class PresetsController {
    private final @NotNull PresetsView view;

    public PresetsController(
        final @NotNull PresetsView view,
        final @NotNull TextEditingArea textEditingArea) {
        this.view = view;
        SettingsThemePresets.THEMES.forEach(themeEntry -> {
            final var themeName = themeEntry.name();
            final var settingsTheme = themeEntry.theme();
            final var section = new PresetsView.PresetSection(themeName, settingsTheme);
            section.button.addActionListener(e -> {
                final var theme = settingsTheme.toTheme();
                textEditingArea.setTheme(theme);
                // TODO: sync with base style and syntax style settings
            });
            view.addSection(section);
        });
    }
}

File: src/main/java/rars/venus/settings/editor/controllers/PresetsController.kt
package rars.venus.settings.editor.controllers

import rars.settings.SettingsThemePresets.THEMES
import rars.venus.editors.TextEditingArea
import rars.venus.settings.editor.views.PresetsView
import rars.venus.settings.editor.views.PresetsView.PresetSection

fun PresetsController(
    view: PresetsView,
    textArea: TextEditingArea,
    parentController: EditorSettingsController
) {
    THEMES.forEach { themeEntry ->
        val section = PresetSection(themeEntry.name)
        section.button.addActionListener {
            parentController.settingsTheme = themeEntry.theme.clone()
            textArea.setTheme(parentController.settingsTheme.toEditorTheme())
            // We need to update the info regarding the theme in all the theme
            // related controllers.
            parentController.updateThemeControllers()
        }
        view.addSection(section)
    }
}
